apigw/src/shared/auth/jwt.ts

```diff
@@ -19,10 +20,65 @@ export function createJwt(payload: {
     | 'mobile'
     | 'system'
     | 'integration'
-}): string {
-  return jwt.sign(payload, privateKey, {
-    algorithm: 'RS256',
-    expiresIn: '48h',
-    keyid: jwtKid
+}
+
+export function createJwt(payload: JwtPayload): string {
+  return tracer.trace('createJwt', (span) => {
+    const now = Date.now()
+    const key = cacheKey(payload)
+    const cached = cache.get(key)
+    if (cached && now - cached.createdAt <= JWT_LIFETIME) {
+      span?.addTags({ cacheHit: true, cacheSize: cache.size })
+      return cached.token
+    } else {
+      const token = jwt.sign(payload, privateKey, {
+        algorithm: 'RS256',
+        expiresIn: '48h',
+        keyid: jwtKid
+      })
+      cache.set(key, { token, createdAt: now })
+      span?.addTags({ cacheHit: false, cacheSize: cache.size })
```

Comment:
Alternative way to `CACHE_CLEANUP_INTERVAL` would be a create a timer for each cache entry like:
```
    // Set a timer to remove the cache entry after JWT_LIFETIME
    setTimeout(() => {
        cache.delete(key);
    }, JWT_LIFETIME);
```
this would reduce the code complexity and avoid the need to iterate trough all cache entries.